File: src/com/services/Contractor.java
package com.services;

public interface Contractor {
	int retriveData(String sitename);                    //retrive data using sitename
	int retriveData(String sitename,String date);		//retrive  data using sitename and date
	int deleteData(String sitename);			//delete  data using sitename
	int deleteData(String sitename,String date);		//delete data using sitename and date
	int laborRateChange(int rate);				//change in bebor rate 
	int workerRateChange(int rate);				//change in worker rate
	int cementRateChange(int rate);				//change in cement rate
	int sandRateChange(int rate);				//change in sand rate
	int brickRateChange(int rate);				//change in brick rate
	String insertUser(String sitename,String role);		//inert new user
	int supervisorSiteChange(int username,String newsite);	//change in supervisor supervision site
	int customerSiteChange(int username,String password,String sitename);	//change in customer site in this case customer password is must

}

File: src/com/services/Contractor.go
package services

type Contractor interface {
	RetriveData01(sitename string) int                                     //retrive data using sitename
	RetriveData02(sitename string, date string) int                        //retrive  data using sitename and date
	DeleteData01(sitename string) int                                      //delete  data using sitename
	DeleteData02(sitename string, date string) int                         //delete data using sitename and date
	laborRateChange(rate int) int                                          //change in bebor rate
	workerRateChange(rate int) int                                         //change in worker rate
	cementRateChange(rate int) int                                         //change in cement rate
	sandRateChange(rate int) int                                           //change in sand rate
	brickRateChange(rate int) int                                          //change in brick rate
	insertUser(sitename string, role string) string                        //inert new user
	supervisorSiteChange(username int, newsite string) int                 //change in supervisor supervision site
	customerSiteChange(username int, password string, sitename string) int //change in customer site in this case customer password is must

}
